src/main/kotlin/org/rust/ide/navigation/goto/RustGotoSuperHandler.kt

```diff
@@ -4,20 +4,36 @@ import com.intellij.lang.LanguageCodeInsightActionHandler
 import com.intellij.openapi.editor.Editor
 import com.intellij.openapi.project.Project
 import com.intellij.psi.PsiFile
+import org.rust.lang.core.psi.RustImplItemElement
+import org.rust.lang.core.psi.RustImplMethodMemberElement
 import org.rust.lang.core.psi.RustMod
+import org.rust.lang.core.psi.RustNamedElement
 import org.rust.lang.core.psi.impl.RustFile
 import org.rust.lang.core.psi.util.parentOfType
+import org.rust.lang.core.psi.util.trait
 
 class RustGotoSuperHandler : LanguageCodeInsightActionHandler {
     override fun startInWriteAction() = false
 
     override fun invoke(project: Project, editor: Editor, file: PsiFile) {
         val focusedElement = file.findElementAt(editor.caretModel.offset) ?: file ?: return
 
-        val focusedMod = focusedElement.parentOfType<RustMod>(strict = false) ?: return
-        val superMod = focusedMod.`super` ?: return
+        val focusedMethod = focusedElement.parentOfType<RustImplMethodMemberElement>(strict = false)
```

Comment:
Technically, we should search *simultaneously* for both methods and modules: 

```
impl Foo for Bar {
     fn foo() {
        mod m {  mod n { <caret> /* ideally, should find n and not foo */ } }
     }
}
```

Though I think this is an obscure corner case, so perhaps there is no need in handling it.